File: src/app/shared/models/item-cardapio.ts
import { Pedido } from "./pedido";

export class ItemCardapio {
  id: number;
  nome: string;
  tipo: string;
  descricao: string;
  preco: number;
  pedidos: Array<Pedido>;

  constructor(id: number = 0, nome: string = '', descricao: string= '', tipo: string = '', preco: number = 0){
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.tipo = tipo;
    this.preco = preco;
    this.pedidos = new Array<Pedido>();
  }
}

File: Projeto Spring Boot/src/main/java/br/edu/ifpb/gugawag/trazaconta/model/ItemCardapio.java
package br.edu.ifpb.gugawag.trazaconta.model;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Currency;
import java.util.List;

@Entity
public class ItemCardapio {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long Id;

    private String nome;
    private String tipo;
    private String descricao;
    private BigDecimal preco;

    @ManyToMany(mappedBy="itensCardapio")
    private List<Pedido> pedidos;

  public Long getId() {
    return Id;
  }

  public void setId(Long id) {
    this.Id = id;
  }

  public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public BigDecimal getPreco() {
        return preco;
    }

    public void setPreco(BigDecimal preco) {
        this.preco = preco;
    }
}
